sw/device/tests/sim_dv/rv_dm_access_after_wakeup.c

```diff
@@ -0,0 +1,141 @@
+                                    cfg);
+  LOG_INFO("%s", deep_sleep ? "Entering deep sleep." : "Entering normal sleep.");
+  wait_for_interrupt();
+}
+
+bool test_main(void) {
+  // Enable global and external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+
+  dif_pinmux_t pinmux;
+  dif_pwrmgr_t pwrmgr;
+  dif_sysrst_ctrl_t sysrst_ctrl;
+
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+  CHECK_DIF_OK(dif_rv_plic_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR), &rv_plic));
+  CHECK_DIF_OK(dif_sysrst_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_SYSRST_CTRL_AON_BASE_ADDR),
+      &sysrst_ctrl));
+
+  switch (rstmgr_testutils_reason_get()) {
+    case kDifRstmgrResetInfoPor: // The first power-up.
+      LOG_INFO("Software Setup.");
+      // Wait for sequence to run its checks.
+      IBEX_SPIN_FOR(kSoftwareBarrier == 1, kSoftwareBarrierTimeoutUsec);
+
+      // Enable all the AON interrupts used in this test.
+      rv_plic_testutils_irq_range_enable(&rv_plic, kTopEarlgreyPlicTargetIbex0,
+                                         kTopEarlgreyPlicIrqIdPwrmgrAonWakeup,
+                                         kTopEarlgreyPlicIrqIdPwrmgrAonWakeup);
+
+      // Enable pwrmgr interrupt.
+      CHECK_DIF_OK(dif_pwrmgr_irq_set_enabled(&pwrmgr, kDifPwrmgrIrqWakeup,
+                                              kDifToggleEnabled));
+
+      // Set up power button as wake up source.
+      dif_sysrst_ctrl_input_change_config_t config = {
+          .input_changes = kDifSysrstCtrlInputPowerButtonH2L,
+          .debounce_time_threshold = 1,  // 5us
+      };
+      CHECK_DIF_OK(
+          dif_sysrst_ctrl_input_change_detect_configure(&sysrst_ctrl, config));
+      CHECK_DIF_OK(dif_pinmux_input_select(
+          &pinmux, kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonPwrbIn,
+          kTopEarlgreyPinmuxInselIor13));
+
+      // Put the device in a normal sleep.
+      put_to_sleep(&pwrmgr, /*deep_sleep=*/false);
+      LOG_INFO("Waking up from normal sleep.");
+
+      // Clean up wakeup source after sleep.
+      CHECK_DIF_OK(dif_sysrst_ctrl_ulp_wakeup_clear_status(&sysrst_ctrl));
+
+      // Wait for sequence to run its checks.
+      IBEX_SPIN_FOR(kSoftwareBarrier == 2, kSoftwareBarrierTimeoutUsec);
+
+      // Put the device in a deep sleep.
+      put_to_sleep(&pwrmgr, /*deep_sleep=*/true);
+      break;
+
+    case kDifRstmgrResetInfoLowPowerExit: // The power up after deep sleep.
+      LOG_INFO("Waking up from deep sleep.");
+
+      // Wait for sequence to finish before returning.
+      IBEX_SPIN_FOR(kSoftwareBarrier == 3, kSoftwareBarrierTimeoutUsec);
+      return true;
+
+    default:
+      CHECK(false, "Device was reset by an unexpected source.");
+      break;
```

Comment:
This feels equivalent to `LOG_ERROR` followed by `return false`, which is perhaps a tad bit easier to read?

Reply:
Yeah, I agree. I'll change this once I have fixed the problem below.

Reply:
I will take a look at the failure later tonight.

Reply:
Thanks! Thankfully the test is passing again. I've also changed to use `LOG_ERROR` for unexpected reset sources.

Reply:
@sriyerg I think it is passing again - see timestamp of @HU90m's message above.